File: frontend/src/app/models/task.ts
export type TaskStatus = 'TO_DO' | 'IN_PROGRESS' | 'DONE';

export interface Task {
  id: number;
  title: string;
  description?: string | null;
  status: TaskStatus;
  createdAt: string; // ISO string
}

export interface TaskCreateRequest {
  title: string;
  description?: string | null;
  status: TaskStatus;
}

export interface TaskUpdateRequest {
  title: string;
  description?: string | null;
  status: TaskStatus;
}

File: backend/src/main/java/com/example/taskmanager/task/Task.java
package com.example.taskmanager.task;
import com.example.taskmanager.task.TaskStatus;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "tasks")
public class Task {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(nullable = false, length = 120)
  private String title;

  @Column(columnDefinition = "TEXT")
  private String description;

  @Enumerated(EnumType.STRING)
  @Column(nullable = false, length = 20)
  private TaskStatus status;

  @CreationTimestamp
  @Column(nullable = false, updatable = false)
  private LocalDateTime createdAt;

  @Column(nullable = false)          // ← ADD THIS
  private Long userId;               // ← ADD THIS
}